src/hotspot/share/classfile/moduleEntry.hpp

```diff
@@ -185,6 +185,10 @@ class ModuleEntry : public CHeapObj<mtModule> {
   static ModuleEntry* create_boot_unnamed_module(ClassLoaderData* cld);
   static ModuleEntry* new_unnamed_module_entry(Handle module_handle, ClassLoaderData* cld);
 
+  // Note caller requires ResourceMark
+  const char* name_as_C_string() {
+    return is_named() ? name()->as_C_string() : UNNAMED_MODULE;
+  }
```

Comment:
Maybe PackageEntry::name_as_C_string() should get the same helper function.

Reply:
Can't do that. The un-named package means you have no PackageEntry and package() returns null.

Reply:
Ok.